echo2_config.cc

```diff
@@ -16,8 +16,20 @@ class Echo2ConfigFactory : public NamedNetworkFilterConfigFactory {
 public:
   // NamedNetworkFilterConfigFactory
   NetworkFilterFactoryCb createFilterFactory(const Json::Object&, FactoryContext&) override {
-    return [](Network::FilterManager& filter_manager)
-        -> void { filter_manager.addReadFilter(Network::ReadFilterSharedPtr{new Filter::Echo2()}); };
+    return [](Network::FilterManager& filter_manager) -> void {
```

Comment:
Do we still need v1 JSON here? Best to start purging this from the examples.

Reply:
Done at 6368076. To clarify, this incremental update of `echo2` test is actually to support https://github.com/envoyproxy/envoy/pull/2894, i.e. to remove the ci noise regarding this repo.